context/sm_context.go

```diff
@@ -421,12 +420,8 @@ func (smContext *SMContext) PCFSelection() error {
 
 		if res != nil {
 			if status := res.StatusCode; status != http.StatusOK {
-				metrics.IncrementSvcNrfMsgStats(SMF_Self().NfInstanceID, string(svcmsgtypes.NnrfNFDiscoveryPcf), "In", "Failure", err.Error())
-				apiError := err.(openapi.GenericOpenAPIError)
-				problemDetails := apiError.Model().(models.ProblemDetails)
```

Comment:
Why are these getting deleted?

Reply:
From line 412 we know that `err == nil` 
so it would be a nil pointer dereference  in `err.Error()` (line 424) and `err.(openapi.GenericOpenAPIError)`  (line 425)